src/responder/nss/nsssrv.c

```diff
@@ -124,6 +124,47 @@ nss_clear_memcache(TALLOC_CTX *mem_ctx,
     return ret;
 }
 
+static errno_t
+nss_clear_negcache(TALLOC_CTX *mem_ctx,
+                   struct sbus_request *sbus_req,
+                   struct nss_ctx *nctx)
+{
+    errno_t ret;
+
+    // if (access(SSS_NSS_MCACHE_DIR"/"CLEAR_NC_FLAG, F_OK) < 0) {
+    //     ret = errno;
+    //     if (ret == ENOENT) {
+    //         DEBUG(SSSDBG_TRACE_FUNC,
+    //               "CLEAR_NC_FLAG not found. Nothing to do.\n");
+    //         return ret;
+    //     } else {
+    //         DEBUG(SSSDBG_CRIT_FAILURE,
+    //               "Failed to check existence of "CLEAR_NC_FLAG": %s.\n",
+    //               strerror(ret));
+    //         return ret;
+    //     }
+    // }
+
+    DEBUG(SSSDBG_TRACE_FUNC, "NSS negative cache reset and repopulate \
+                              with permament entries\n");
+
+    ret = sss_ncache_reset_repopulate_permanent(nctx->rctx,
```

Comment:
IIUC, this function only removes permanent entries, i.e. does the opposite of required.

Reply:
I initially used it due to description in negcache.h:
```
/* Flush the negcache and then repopulate */
errno_t sss_ncache_reset_repopulate_permanent(struct resp_ctx *rctx, struct sss_nc_ctx *ncache);
```

But taking a look at the details:
```
/* Reset permanent negcache after checking the domains */
errno_t sss_ncache_reset_repopulate_permanent(struct resp_ctx *rctx,
                                              struct sss_nc_ctx *ncache)
{
    int ret;

    ret = sss_ncache_reset_permanent(ncache);
    if (ret == EOK) {
        ret = sss_ncache_prepopulate(ncache, rctx->cdb, rctx);
    }

    return ret;
}
```
So you are right. I will add missing `sss_ncache_reset_users()` and `sss_ncache_reset_groups()` calls.

Reply:
What is the reason to touch permanent entries (sss_ncache_reset_repopulate_permanent()) here?